### PR TITLE
Unit test

### PR DESCRIPTION
**Pull request by Cosmosha**
### The following requirement and features were added;

- Add Testing for Book
- Add Testing for Person
- Add Testing for Student
- Add Testing for Teacher
- Add Testing for Rental
- Add Testing for Classroom
- All Decorator Classes

### CODE REVIEW

File: spec/base_decorator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require './base_decorator'
+
+describe BaseDecorator do
```
Comment (Felix45) [inline spec/base_decorator_spec.rb:3]: - [ Required ] Please consider adding a unit test for the `correct_name` method in the `BaseDecorator` class. One of the project requirements is testing all the methods in our classes including those created while implementing the Decorator design pattern. Please see the screenshot below and rectify this.

## Project requirements
Screenshot from 2023-06-16 18-38-05

File: spec/capitalize_decorator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require './capitalize_decorator'
+require './nameable'
+describe CapitalizeDecorator do
```
Comment (Felix45) [inline spec/capitalize_decorator_spec.rb:3]: - [ Required ] Please consider adding a unit test for the `correct_name` method in the `CapitalizeDecorator` class. One of the project requirements is testing all the methods in our classes including those created while implementing the Decorator design pattern. Please see the screenshot below and rectify this.

## Project requirements
Screenshot from 2023-06-16 18-38-05

File: spec/trimmer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './trimmer_decorator'
+require './nameable'
+describe TrimmerDecorator do
```
Comment (Felix45) [inline spec/trimmer_decorator.rb:3]: - [ Required ] Please consider adding a unit test for the `correct_name` method in the `TrimmerDecorator` class. One of the project requirements is testing all the methods in our classes including those created while implementing the Decorator design pattern. Please see the screenshot below and rectify this.

## Project requirements
Screenshot from 2023-06-16 18-38-05

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @Cosmosha & @yosaddis, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :heavy_check_mark: 
- [x] Rspec testing gem is used correctly :white_check_mark: 
- [x] All unit tests pass with green check marks :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Good descriptive commit messages :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:    


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Cosmosha & @yosaddis, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Unit Testing done on all the files ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**